observability/src/test/java/io/grpc/observability/interceptors/ConfigFilterHelperTest.java

```diff
@@ -0,0 +1,229 @@
+        .collect(Collectors.toList());
+    ImmutableList<LogFilter> sampleLogFilters =
+        ImmutableList.copyOf(sampleFilters);
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.getLogFilters()).thenReturn(sampleLogFilters);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+
+    String fullMethodName = "service1/Method6";
+    MethodFilterParams expectedParams =
+        configFilterHelper.globalParams;
+    MethodFilterParams resultParams
+        = configFilterHelper.isMethodToBeLogged(fullMethodName);
+
+    assertEquals(resultParams.log, expectedParams.log);
+    assertEquals(resultParams.headerBytes, expectedParams.headerBytes);
+    assertEquals(resultParams.messageBytes, expectedParams.messageBytes);
+  }
+
+  @Test
+  public void checkMethodNotToBeLogged() {
+    List<LogFilter> sampleFilters = Stream.of(
+            new LogFilter("service1/Method2", 1024, 1024),
+            new LogFilter("service2/*", 2048, 1024))
+        .collect(Collectors.toList());
+    ImmutableList<LogFilter> sampleLogFilters =
+        ImmutableList.copyOf(sampleFilters);
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.getLogFilters()).thenReturn(sampleLogFilters);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+
+    String fullMethodName = "service3/Method3";
+    MethodFilterParams expectedParams =
+        new MethodFilterParams(false, 0, 0);
+    MethodFilterParams resultParams
+        = configFilterHelper.isMethodToBeLogged(fullMethodName);
+
+    assertEquals(resultParams.log, expectedParams.log);
+    assertEquals(resultParams.headerBytes, expectedParams.headerBytes);
+    assertEquals(resultParams.messageBytes, expectedParams.messageBytes);
+  }
+
+  @Test
+  public void checkMethodToBeLoggedConditional() {
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.getLogFilters()).thenReturn(configLogFilters);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+
+    String fullMethodName = "service1/Method2";
+    MethodFilterParams expectedParams =
+        new MethodFilterParams(true, 1024, 1024);
+    MethodFilterParams resultParams
+        = configFilterHelper.isMethodToBeLogged(fullMethodName);
+
+    assertEquals(resultParams.log, expectedParams.log);
+    assertEquals(resultParams.headerBytes, expectedParams.headerBytes);
+    assertEquals(resultParams.messageBytes, expectedParams.messageBytes);
+
+    String serviceNameWildcard = "service2/*";
+    MethodFilterParams expectedParamsWildCard =
+        new MethodFilterParams(true, 2048, 1024);
+    MethodFilterParams resultParamsWildCard
+        = configFilterHelper.isMethodToBeLogged(serviceNameWildcard);
+
+    assertEquals(resultParamsWildCard.log, expectedParamsWildCard.log);
+    assertEquals(resultParamsWildCard.headerBytes, expectedParamsWildCard.headerBytes);
+    assertEquals(resultParamsWildCard.messageBytes, expectedParamsWildCard.messageBytes);
+  }
+
+  @Test
+  public void checkEventToBeLogged() {
```

Comment:
Are we testing code that is not actually used in the main code? We are not doing event based filtering?

Reply:
Added event based filtering.